perfkitbenchmarker/flag_util.py

```diff
@@ -190,3 +191,149 @@ def DEFINE_integerlist(name, default, help, on_nonincreasing=None,
   serializer = IntegerListSerializer()
 
   flags.DEFINE(parser, name, default, help, flag_values, serializer, **args)
+
+
+SI_SUFFIXES = ['', 'K', 'M', 'G', 'T', 'P']
+
+
+@functools.total_ordering
+class ObjectSize(object):
+  """Holds the size of data, i.e. numbers of bytes.
+  """
+
+  def __init__(self, bytes=0):
+    self.bytes = bytes
+
+  def mult_for_base_and_exp(self, base, exp):
+    """Given base and exp, returns mult so that
+    bytes = mult * base ^ exp.
+
+    May return a float.
+    """
+
+    base_to_exp = base ** exp
+    return float(self.bytes) / base_to_exp
+
+  def exact_mult_for_base_and_exp(self, base, exp):
+    """Given base and exp, returns mult so that
+    bytes = mult * base ^ exp.
+
+    Raises ValueError if mult is not an integer.
+    """
+
+    base_to_exp = base ** exp
+    if self.bytes % base_to_exp == 0:
+      return self.bytes / base_to_exp
+    else:
+      raise ValueError('Bytes %s not divisible by %s',
+                       self.bytes, base_to_exp)
+
+  def ExactB(self):
+    return self.bytes
+
+  def ExactKiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 1)
+
+  def ExactMiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 2)
+
+  def ExactGiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 3)
+
+  def AsB(self):
```

Comment:
Considering bytes is a public attribute, this method doesn't seem necessary, but the rest should be consistent. Consider making these methods properties or making bytes private.